app/lib/system/error_reporting.rb

```diff
@@ -12,6 +12,17 @@ def report_error(exception, logger: Rails.logger, **parameters)
       end
     end
 
+    def report_deprecation_warning(payload)
+      message = payload[:message]
+      deprecation_horizon = payload[:deprecation_horizon]
+
+      ::Bugsnag.notify(message) do |report|
```

Comment:
I like how in [this article](https://www.bugsnag.com/blog/managing-warnings-in-ruby) they wrap the message in `Deprecation` class, I guess what it does that in BugSnag it will appear as Deprecation, and not as RuntimeError?
I think it would be nice.

Reply:
I pushed some changes, now it reports the error as `DeprecationWarning`. That's very useful to filter by `errorClass:DeprecationWarning` from Bugsnag.

For the record, just in case somebody finds this in the future: there are two more error classes called `DeprecationError`, those have their own path to reach Bugsnag and seem to be used to report a different kind of deprecations: when we deprecate a liquid tag or method that we provide for clients to use in their developer portals. I was considering to reuse those exceptions, but the use case is different so I decided to just send a String to Bugsnag, in order to not make those `DeprecationError` more confuse. Now I'm adding a third exception called `DeprecationWarning` and we use this only for development purposes. So summarizing:

`DeprecationWarning` -> Only for us.
`DeprecationError` -> For developer portals.